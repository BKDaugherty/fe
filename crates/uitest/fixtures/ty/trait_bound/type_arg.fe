pub struct Foo<T: Trait1 + Trait2> {
    t: T
}

trait Trait1 {}
trait Trait2 {}
impl Trait1 for i32 {}

pub struct User {
    t: Foo<i32>,
    u: Foo<u32>
}