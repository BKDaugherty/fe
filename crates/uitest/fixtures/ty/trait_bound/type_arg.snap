---
source: crates/uitest/tests/ty.rs
expression: diags
input_file: crates/uitest/fixtures/ty/trait_bound/type_arg.fe
---
error[6-0003]: trait bound is not satisfied
   ┌─ type_arg.fe:10:8
   │
10 │     t: Foo<i32>,
   │        ^^^^^^^^ `i32` doesn't implement `Trait2`

error[6-0003]: trait bound is not satisfied
   ┌─ type_arg.fe:11:8
   │
11 │     u: Foo<u32>
   │        ^^^^^^^^ `u32` doesn't implement `Trait1`


